From 61e0df50bc4954402998c7caa94a95cb660478f2 Mon Sep 17 00:00:00 2001
From: "Badrappan, Jeevaka" <jeevaka.badrappan@intel.com>
Date: Mon, 24 Jul 2023 08:50:06 +0530
Subject: [PATCH] [WA] Disable m_blokCopyon to address crash in KOG

KOG app is crashing in demo mode.

Crash call stack:
MhwBltInterfaceXe_Hp_Base::AddBlockCopyBlt(_MOS_COMMAND_BUFFER*, _MHW_FAST_COPY_BLT_PARAM*, unsigned int, unsigned int)+1713)
BltState_Xe_Hpm::SubmitCMD(_BLT_STATE_PARAM*)+1028)
BltState_Xe_Hpm::CopyMainSurface(_MOS_SPECIFIC_RESOURCE*, _MOS_SPECIFIC_RESOURCE*)+111)
MediaCopyState_Xe_Hpm::MediaBltCopy(_MOS_SPECIFIC_RESOURCE*, _MOS_SPECIFIC_RESOURCE*)+46)

Issue is seen from the "[Media Common] [VP] enable block copy for DG2"
commit that got introduced from intel-media-23.2.3.

Disabled block copy for DG2 as a workaround.

Change-Id: Ic98bd1d036783dabca1b18c2bade9f882b56a8a0
Tracked-On: OAM-111325
Signed-off-by: Badrappan, Jeevaka <jeevaka.badrappan@intel.com>
---
 .../Xe_M/Xe_HPM/shared/mediacopy/media_blt_copy_xe_hpm.cpp    | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/media_driver/media_softlet/agnostic/Xe_M/Xe_HPM/shared/mediacopy/media_blt_copy_xe_hpm.cpp b/media_driver/media_softlet/agnostic/Xe_M/Xe_HPM/shared/mediacopy/media_blt_copy_xe_hpm.cpp
index 76ed12e44..d2cb18f27 100644
--- a/media_driver/media_softlet/agnostic/Xe_M/Xe_HPM/shared/mediacopy/media_blt_copy_xe_hpm.cpp
+++ b/media_driver/media_softlet/agnostic/Xe_M/Xe_HPM/shared/mediacopy/media_blt_copy_xe_hpm.cpp
@@ -97,7 +97,7 @@ BltState_Xe_Hpm::~BltState_Xe_Hpm()
 MOS_STATUS BltState_Xe_Hpm::Initialize()
 {
     initialized  = true;
-    m_blokCopyon = true;
+    m_blokCopyon = false;
     return MOS_STATUS_SUCCESS;
 }
 
@@ -803,4 +803,4 @@ MOS_STATUS BltState_Xe_Hpm::CopyMainSurface(
 
     MOS_TraceEventExt(EVENT_MEDIA_COPY, EVENT_TYPE_END, nullptr, 0, nullptr, 0);
     return MOS_STATUS_SUCCESS;
-}
\ No newline at end of file
+}
-- 
2.40.0

